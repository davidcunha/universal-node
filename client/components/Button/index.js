import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './styles.css';

/**
 * Button
 */
class Button extends Component {
  static propTypes = {
    /**
     * Class name for the Button wrapper container
     */
    wrapperClassname: PropTypes.string,
    /**
     * Class name for the Button container
     */
    className: PropTypes.string,
    /**
     * Text that appears inside the button
     */
    text: PropTypes.string.isRequired,
    /**
     * Name to be given to identify the button element
     */
    name: PropTypes.string.isRequired,
    /**
     * The type of button that needs to be rendered,
     * that by default is assumed to be "button"
     */
    type: PropTypes.string,
    /**
     * Action that will be called when the button is clicked
     */
    action: PropTypes.func,
    /**
     * Boolean attribute that specifies if the button element
     * has been selected by an outside action
     */
    selected: PropTypes.bool,
    /**
     * Identifying id of the itme surrounded by the button
     */
    dataId: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    /**
     * Flag for the parent action to know whether
     * or not to call another related function
     */
    dataCallback: PropTypes.string,
    /**
     * Boolean attribute that specifies a second button style form
     */
    disabled: PropTypes.bool,
  };

  render() {
    const {
      wrapperClassname,
      className,
      name,
      type,
      text,
      action,
      selected,
      dataId,
      dataCallback,
      disabled,
    } = this.props;

    return (
      <div className={classNames(styles['submit-wrapper'], wrapperClassname)} data-id={dataId}>
        <div className={classNames(styles['submit-wrapper-relative'])} data-id={dataId}>
          <button
            id={dataId}
            className={classNames(styles.button, selected && styles.selected, className)}
            name={name}
            type={type || 'button'}
            onClick={action}
            data-id={dataId}
            data-callback={dataCallback}
            disabled={disabled}
          >
            {text}
          </button>
        </div>
      </div>
    );
  }
}

export default Button;
