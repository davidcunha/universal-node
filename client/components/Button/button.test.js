/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import Button from '~/components/Button';

describe('Button', () => {
  it('should render an empty button', () => {
    const component = shallow(<Button text="test" />);

    expect(component.getElements()).toMatchSnapshot();
  });
});
