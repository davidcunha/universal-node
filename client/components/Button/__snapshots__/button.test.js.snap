// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render an empty button 1`] = `
Array [
  <div
    className="submit-wrapper"
  >
    <div
      className="submit-wrapper-relative"
    >
      <button
        className="button"
        type="button"
      >
        test
      </button>
    </div>
  </div>,
]
`;
